doc/contribute/nest_with_ides.rst

```diff
@@ -0,0 +1,103 @@
+#. In Eclipse preferences, under ``General > Security > Secure Storage > Advanced``
+   you should replace the default password encryption scheme to a more secure level
+   than default, e.g. to ``PBE...SHA512...AES_256``.
+
+Setting up the project
+~~~~~~~~~~~~~~~~~~~~~~
+
+#. Clone NEST onto your computer
+#. Build NEST manually
+
+   a. Create a build directory
+   #. Run ``cmake`` with suitable options
+   #. Run ``make all`` and ``make install``
+#. In Eclipse
+
+   a. choose ``New > Makefile project with existing code``
+   #. select the directory containing the NEST source code
+   #. choose, e.g., the ``Cross GCC Toolchain``
+#. Right click the project and choose ``Properties`` from the context
+   menu
+
+   a. Under ``C/C++ Build/Build Variables`` define ``BUILD_DIR`` and ``CONDA_ENV``,
+      both of type ``Path``. The first should contain the full path to the build
+      directory you created above, the second the full path to your conda 
+      environment, usually something like ``.../miniconda3/envs/nest-dev``.
+   #. Under ``C/C++ Build > Builder Settings``
+
+      #. uncheck ``Use default build command``
+      #. set ``Build Command`` to ``make -k -j4 all install`` (adjust
+	 number of processes to your situation)
+      #. set ``Build Directory`` to ``${BUILD_DIR}``
+   #. Under ``C/C++ Build > Environment`` prepend
+      ``${CONDA_ENV}/bin`` to ``PATH``
+   #. Under ``C/C++ General > Paths and Symbols > Includes`` add the
+      following two direcories
+      * ``${BUILD_DIR}/libnestutil`` (contains ``config.h``)
+      * ``${CONDA_ENV}/include`` (all headers from packages provided in conda environment)
+   #. Under ``PyDev - Interpreter/Grammar`` choose the intepreter from
+      your Conda environment (you may need to add it by following the
+      ``Click here to configure an interpreter not listed`` link and
+      then ``Browse for python/pypy exe`` (this temporarily takes you
+      to the global Eclipse preferences in a separate window).
+   #. Under ``Run/Debug Settings``, add a ``New ...`` launch
+      configuration, entering in ``C/C++ Application`` the full path
+      to the installed ``nest`` executable.
+
+What you can do then
+~~~~~~~~~~~~~~~~~~~~
+
+* Eclipse should now find all includes.
+* It should intepret switches such as ``HAVE_GSL`` and ``HAVE_MPI``
+  correctly, i.e., shade the code for the option that is not given.
+  If this does not seem to work, try to rebuild the C/C++ index by
+  opening C++ source file and chosing ``Project > C/C++ Index >
+  Rebuild``.
+* Clicking the hammer icon should compile and install NEST. Errors
+  will be shown in the console and summarized in the warnings tab
+  and you can jump directly to corresponding code locations.
+* ``Run`` should run NEST in a console inside Eclipse. Under Linux,
+  ``Debug`` should also start a debugging session. To get most out of
+  debugging, run ``cmake`` in the build directory with
+  ``-Dwith-debug=ON``.
```

Comment:
I wonder if these bullet points could be turned into subsections, e.g. "Running NEST".

Reply:
I think the text for each of the bullet points is so short that subsections seem a bit over the top.

Reply:
Fair enough.